### PR TITLE
feat(db): database migrations are now managed by phinx

### PR DESCRIPTION
**Pull request by mrclay**
Rebased and adapted #11055

Added: `chore(cache): no more warnings for missing cache dirs`

### CODE REVIEW

File: engine/classes/Elgg/Application.php [hunk 1]
```diff
@@ -647,6 +651,62 @@ public static function upgrade() {
 	}
 
 	/**
+	 * Runs database migrations
+	 *
+	 * @throws InstallationException
+	 * @return bool
+	 */
+	public static function migrate() {
```
Comment (hypeJunction) [inline engine/classes/Elgg/Application.php]: We should probably move settings config migrations here as well

**Comment by hypeJunction** [comment]
Travis says: Could not determine database name! Please specify a database name in your config file.

**Comment by mrclay** [comment]
Ahh, during install, DB details aren't in settings. Later they are.

**Comment by mrclay** [comment]
@hypeJunction I need the migrator to add these (for now at least): https://github.com/Elgg/Elgg/commit/14f5dd0e69e31c4f5763cc1ab9c6d4e7c706e47e#diff-9b29e76c680b51712a924f15eb0b245b

File: engine/schema/migrations/20170728074729_create_entity_subtypes_table.php [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+		]);
+
+		$table->save();
+
+		$table->insert([
+			[
+				'type' => 'object',
+				'subtype' => 'plugin',
+				'class' => 'ElggPlugin',
+			],
```
Comment (mrclay) [inline engine/schema/migrations/20170728074729_create_entity_subtypes_table.php]: @hypeJunction is it OK to add these here for the time being?

**Pull request merged**